### PR TITLE
feat: domain name service

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-99

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/Solana/src/apis/getDomainName.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { clusterApiUrl, Connection, PublicKey } from '@solana/web3.js'
+
+const SOL_TLD_AUTHORITY = new PublicKey('58PwtjSDuFHuUkYjH9BYnnQKHfwo9reZhC2zMJv9JPkx')
+const getKey = async (name: string) => {
+    const hashedName = await getHashedName(name)
+    const domainKey = await getNameAccountKey(hashedName, undefined, SOL_TLD_AUTHORITY)
+    return { domainKey, hashedName }
+}
+
+const connection = (url = clusterApiUrl('mainnet-beta')) => {
```
Comment (guanbinrui) [inline packages/plugins/Solana/src/apis/getDomainName.ts]: You can use `ChainId` of Solana here.

File: cspell.json [hunk 2]
```diff
@@ -39,6 +39,7 @@
         "bluebar",
         "boba",
         "bolivarian",
+        "bonfida",
```
Comment (septs) [inline cspell.json:42]: @Jack-Works can ignore `package.json` file?

Comment (Jack-Works) [reply cspell.json:42]: yeah please

File: packages/plugins/Solana/package.json [hunk 3]
```diff
@@ -4,6 +4,8 @@
   "main": "./dist/index.js",
   "types": "./dist/index.d.ts",
   "dependencies": {
+    "@bonfida/name-offers": "^0.0.1",
+    "@bonfida/spl-name-service": "^0.1.27",
```
Comment (guanbinrui) [inline packages/plugins/Solana/package.json:8]: What's the difference between `@bonfida/spl-name-service` and `@solana/spl-name-service`?

File: packages/plugins/Solana/src/apis/getDomainName.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { performReverseLookup } from '@bonfida/spl-name-service'
+import type { ChainId } from '@masknet/web3-shared-solana'
+import { getHashedName, getNameAccountKey, NameRegistryState } from '@solana/spl-name-service'
+
+import { Connection, PublicKey } from '@solana/web3.js'
+import { NETWORK_ENDPOINTS } from '../constants'
+
+const SOL_TLD_AUTHORITY = new PublicKey('58PwtjSDuFHuUkYjH9BYnnQKHfwo9reZhC2zMJv9JPkx')
```
Comment (guanbinrui) [inline packages/plugins/Solana/src/apis/getDomainName.ts]: Does it need to move to constants?

File: packages/plugins/Solana/package.json (hunk 3 above)
Comment (lelenei) [reply packages/plugins/Solana/package.json:8]: reverse domain api in @bonfida/spl-name-service

**Comment by guanbinrui** [comment]
@lelenei Plz resolve those conflicts.

**Pull request merged**